homeassistant/components/sensor/rmvtransport.py

```diff
@@ -0,0 +1,210 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_NEXT_DEPARTURE = 'nextdeparture'
+
+CONF_STATION = 'station'
+CONF_DESTINATIONS = 'destinations'
+CONF_DIRECTIONS = 'directions'
+CONF_LINES = 'lines'
+CONF_PRODUCTS = 'products'
+CONF_TIMEOFFSET = 'timeoffset'
+CONF_MAXJOURNEYS = 'max'
+
+DEFAULT_NAME = 'RMV Journey'
+
+VALID_PRODUCTS = ['U-Bahn', 'Tram', 'Bus', 'S', 'RB', 'RE', 'EC', 'IC', 'ICE']
+
+ICONS = {
+    'U-Bahn': 'mdi:subway',
+    'Tram': 'mdi:tram',
+    'Bus': 'mdi:bus',
+    'S': 'mdi:train',
+    'RB': 'mdi:train',
+    'RE': 'mdi:train',
+    'EC': 'mdi:train',
+    'IC': 'mdi:train',
+    'ICE': 'mdi:train',
+    'SEV': 'mdi:checkbox-blank-circle-outline',
+    None: 'mdi:clock'
+}
+ATTRIBUTION = "Data provided by opendata.rmv.de"
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NEXT_DEPARTURE): [{
+        vol.Required(CONF_STATION): cv.string,
+        vol.Optional(CONF_DESTINATIONS, default=['']): cv.ensure_list_csv,
+        vol.Optional(CONF_DIRECTIONS, default=['']): cv.ensure_list_csv,
+        vol.Optional(CONF_LINES, default=['']): cv.ensure_list_csv,
+        vol.Optional(CONF_PRODUCTS, default=VALID_PRODUCTS):
+            vol.All(cv.ensure_list, [vol.In(VALID_PRODUCTS)]),
+        vol.Optional(CONF_TIMEOFFSET, default=0): cv.positive_int,
+        vol.Optional(CONF_MAXJOURNEYS, default=5): cv.positive_int,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string}]
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the RMV departure sensor."""
+    sensors = []
+    for nextdeparture in config.get(CONF_NEXT_DEPARTURE):
+        sensors.append(
+            RMVDepartureSensor(
+                nextdeparture.get(CONF_STATION),
+                nextdeparture.get(CONF_DESTINATIONS),
+                nextdeparture.get(CONF_DIRECTIONS),
+                nextdeparture.get(CONF_LINES),
+                nextdeparture.get(CONF_PRODUCTS),
+                nextdeparture.get(CONF_TIMEOFFSET),
+                nextdeparture.get(CONF_MAXJOURNEYS),
+                nextdeparture.get(CONF_NAME)))
+    add_entities(sensors, True)
+
+
+class RMVDepartureSensor(Entity):
+    """Implementation of an RMV departure sensor."""
+
+    def __init__(self, station, destinations, directions,
+                 lines, products, timeoffset, maxjourneys, name):
+        """Initialize the sensor."""
+        self._station = station
+        self._name = name
+        self.data = RMVDepartureData(station, destinations, directions,
+                                     lines, products, timeoffset, maxjourneys)
+        self._state = STATE_UNKNOWN
+        self._icon = ICONS[None]
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._state != STATE_UNKNOWN
```

Comment:
Don't use `STATE_UNKNOWN`, use `None`.
```py
return self._state is not None
```
Is there no use to differ between unknown state and unavailable?

Reply:
I'll check that.